lib/vault/client.rb

```diff
@@ -325,5 +336,36 @@ def error(response)
 
       raise HTTPError.new(address, response.code, [response.body])
     end
+
+    # Execute the given block with retries and exponential backoff.
+    def with_retries(rescued = [Exception], &block)
+      exception = nil
+
+      backoff_base = self.retry_base
+      backoff_max  = self.retry_timeout
+
+      self.retry_attempts.times do |attempt|
```

Comment:
If I wanted to disable retry behavior, I'd guess setting `retry_attempts` to `0` would do that, but in this case it will cause requests to silently not be made.

Seems like `retry_attempts = 0` should be the thing that makes zero attempts.